---
title: "Troubleshooting Imputation Filering"
author: "Peter Fiorica"
date: "9/5/2018"
output: html_document
---

As of right now, the imputation filtering pipeline using the plink commands and the python script is not working for 1000G and HRC.

For plink, the error is that there are duplicate IDs that do not get removed by --exclude plink.dupvar
For python, the error is:
#  File "UMich_vcf2pxfixCAAPA.py", line 71, in <module>
#      (rs, pos, a0, a1) = arr[0].split(":")
#  ValueError: too many values to unpack

The plink commands seem to be faster and cleaner than the python script, but a third alternative option is to use vcftools.
This option could be used in conjunction with plink to remove duplicates early, and filter by r2 and maf in plink

The generic plink commands are as follows:
```{bash}
plink --vcf /home/peter/AA_nonGAIN_SCZ/Imputation/UMichResults/1000G/UMich1000G/chr22.dose.vcf.gz --biallelic-only --make-bed --out 1000Gc22biallelic
#This makes the umich .vcf file into a set of bfiles and it removed all SNPs with more than 2 alleles
plink --bfile 1000Gc22biallelic --list-duplicate-vars --out 1000Gc22dups
#lists duplicates (variants that share the same allele codes and BP coordinates)
#for chr22 1000G, there are 25 of these duplicated variants in this set
plink --bfile 1000Gc22biallelic --exclude 1000Gc22dups.dupvar --make-bed --out 1000Gc22nodups 
#excludes variants on .dupvar file, but the log suggests otherwise.
#652230 variants loaded from .bim file.
#--exclude: 652166 variants remaining.
#Warning: At least 25 duplicate IDs in --exclude file.

#It looks like 64 variants, not 25 were removed.  I am not sure what happened with the 25 "duplicate IDs"
plink --bfile 1000Gc22nodups --qual-scores /home/peter/AA_nonGAIN_SCZ/Imputation/UMichResults/1000G/UMich1000G/chr22.info 7 1 1 --qual-threshold 0.8 --maf 0.01 --make-bed  --out /home/peter/AA_nonGAIN_SCZ/Imputation/UMichResults/1000G/UMich1000G/UMichFiltered/chr22
#This should make the final bfile set for the chromsome filted by r2 and maf
#Error: Duplicate ID '22:16050654'.
```

9/5/18
An alternative option to this issue is to write a snplist
```{bash}
plink --bfile 1000Gc22biallelic --write-snplist --out all22snps1000G
#This writes us a list of all the SNPs on the chromosome.
cat all22snps1000G.snplist | sort | uniq -d > duplicated22snps1000G.snplist
#This creates a list of duplicates based on the snplist
plink --bfile 1000Gc22biallelic --exclude  duplicated22snps1000G.snplist --make-bed --out 1000Gc22nodups
#this removes the duplicates on the snplist
plink --bfile 1000Gc22nodups --qual-scores /home/peter/AA_nonGAIN_SCZ/Imputation/UMichResults/1000G/UMich1000G/chr22.info 7 1 1 --qual-threshold 0.8 --maf 0.01 --make-bed  --out /home/peter/AA_nonGAIN_SCZ/Imputation/UMichResults/1000G/UMich1000G/UMichFiltered/chr22
#This  makes the final bfile set for the chromsome filted by r2 and maf
```
https://groups.google.com/forum/#!topic/plink2-users/V9Zw2QjkvDI